### PR TITLE
Introduce a test_ci_launcher job

### PR DESCRIPTION
**Pull request by cottsay**
This can be used to validate infrastructure changes that might impact the launcher job.

The job template previously re-computed the downstream job names for predicting the build numbers, but this change re-uses the job names that are passed to the template to begin with, which is probably the right thing to do anyway.

### CODE REVIEW

**Comment by cottsay** [comment]
Here's the diff for the groovy in `ci_launcher`:
```diff
@@ -200,6 +200,6 @@
-build_numbers = [:]
-build_numbers["linux"] = predict_build_number("ci_linux")
-build_numbers["linux-aarch64"] = predict_build_number("ci_linux-aarch64")
-build_numbers["windows"] = predict_build_number("ci_windows")
-
-for (item in build_numbers) {
+predicted_jobs = [:]
+predicted_jobs["linux"] = new Tuple("ci_linux", predict_build_number("ci_linux"))
+predicted_jobs["linux-aarch64"] = new Tuple("ci_linux-aarch64", predict_build_number("ci_linux-aarch64"))
+predicted_jobs["windows"] = new Tuple("ci_windows", predict_build_number("ci_windows"))
+
+for (item in predicted_jobs) {
@@ -210,2 +210,2 @@
-  job_name = "ci_${item.key}"
-  build_number = item.value
+  job_name = item.value[0]
+  build_number = item.value[1]
```

Here's the first invocation of the new `test_ci_launcher` job: https://ci.ros2.org/view/All/job/test_ci_launcher/1/

File: create_jenkins_job.py [hunk 1]
```diff
@@ -551,6 +551,23 @@ def create_job(os_name, job_name, template_file, additional_dict):
...
+            os_specific_data[os_name].update(os_configs[os_name])
+            os_specific_data[os_name]['job_name'] = 'test_ci_' + os_name
+        job_data = dict(data)
+        job_data['ci_scripts_default_branch'] = args.ci_scripts_default_branch
+        job_data['label_expression'] = 'built-in || master'
+        job_data['os_specific_data'] = os_specific_data
+        job_data['cmake_build_type'] = 'None'
+        job_data.update(retention_data_by_job_type(launcher_job_name))
+        job_config = expand_template('ci_launcher_job.xml.em', job_data)
+        configure_job(jenkins, launcher_job_name, job_config, **jenkins_kwargs)
```
Comment (clalancette) [inline create_jenkins_job.py:569]: Given that this code is the same as the above (other than the job prefix), can we combine them?  Something like:

```
for launch_prefix in ('', 'test_'):
    launcher_job_name = prefix + 'ci_launcher'
    if not pattern_select_jobs_regexp or pattern_select_jobs_regexp.match(launcher_job_name):
        os_specific_data = collections.OrderedDict()
        for os_name in sorted(os_configs.keys() - launcher_exclude):
            os_specific_data[os_name] = dict(data)
            os_specific_data[os_name].update(os_configs[os_name])
            os_specific_data[os_name]['job_name'] = prefix + 'ci_' + os_name
        job_data = dict(data)
        job_data['ci_scripts_default_branch'] = args.ci_scripts_default_branch
        job_data['label_expression'] = 'built-in || master'
        job_data['os_specific_data'] = os_specific_data
        job_data['cmake_build_type'] = 'None'
        job_data.update(retention_data_by_job_type(launcher_job_name))
        job_config = expand_template('ci_launcher_job.xml.em', job_data)
        configure_job(jenkins, launcher_job_name, job_config, **jenkins_kwargs)
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: create_jenkins_job.py (hunk 1 above)
Comment (cottsay) [reply create_jenkins_job.py]: Applied nearly verbatim in ad6b945.

**Review by clalancette — APPROVED** [review]

**Comment by cottsay** [comment]
This has been deployed.

**Pull request merged**